client/app/scripts/components/node-details/node-details-image-status.js

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import find from 'lodash/find';
+import map from 'lodash/map';
+import { CircularProgress } from 'weaveworks-ui-components';
+
+import { getImagesForService } from '../../actions/app-actions';
+
+const topologyWhitelist = ['services', 'deployments'];
+
+function getNewImages(images, currentId) {
+  // Assume that the current image is always in the list of all available images.
+  // Should be a safe assumption...
+  const current = find(images, i => i.ID === currentId);
+  const timestamp = new Date(current.CreatedAt);
+  return find(images, i => timestamp < new Date(i.CreatedAt)) || [];
+}
+
+class NodeDetailsImageStatus extends React.PureComponent {
+  constructor(props, context) {
+    super(props, context);
+
+    this.handleServiceClick = this.handleServiceClick.bind(this);
+  }
+
+  componentDidMount() {
+    if (this.shouldRender() && this.props.serviceId) {
+      this.props.getImagesForService(this.props.params.orgId, this.props.serviceId);
+    }
+  }
+
+  handleServiceClick() {
+    const { router, serviceId, params } = this.props;
+    router.push(`/flux/${params.orgId}/services/${encodeURIComponent(serviceId)}`);
+  }
+
+  shouldRender() {
+    const { currentTopologyId } = this.props;
+    return currentTopologyId && topologyWhitelist.includes(currentTopologyId);
+  }
+
+  renderImages() {
+    const { errors, containers, isFetching } = this.props;
+    const error = !isFetching && errors;
+
+    if (isFetching) {
+      return (
+        <div className="progress-wrapper"><CircularProgress /></div>
+      );
+    }
+
+    if (error) {
+      return (
+        <p>Error: {JSON.stringify(map(errors, 'message'))}</p>
+      );
+    }
+
+    if (!containers) {
+      return 'No service images found';
+    }
+
+    return (
+      <div className="images">
+        {containers.map((container) => {
+          const statusText = getNewImages(container.Available, container.Current.ID).length > 0
+            ? <span className="new-image">New image(s) available</span>
+            : 'Image up to date';
+
+          return (
+            <div key={container.Name} className="wrapper">
+              <div className="node-details-table-node-label">{container.Name}</div>
+              <div className="node-details-table-node-label">{statusText}</div>
```

Comment:
Just a tiny styling comment - I'd rename this to `node-details-table-node-value` and apply `text-align: right` to it (`statusText`), think it looks a bit better that way:

![right-styling](https://cloud.githubusercontent.com/assets/1216874/25656855/512eaecc-2ffb-11e7-8466-f6924b63c193.png)